framework/util/image_writer.cpp

```diff
@@ -132,13 +135,88 @@ bool WriteBmpImage(
     return success;
 }
 
+bool WriteBmpImageNoAlpha(
+    const std::string& filename, uint32_t width, uint32_t height, uint64_t data_size, const void* data, uint32_t pitch)
+{
+    bool     success   = false;
+    uint32_t row_pitch = width * kImageBpp;
+
+    if (pitch != 0)
+    {
+        row_pitch = pitch;
+    }
+
+    // BMP image data requires row to be a multiple of 4 bytes
+    // Round-up row size to next multiple of 4, if it isn't already
+    const uint32_t rowSizeNoAlpha = ((width * kImageBppNoAlpha) + 3u) & (~0x03);
+    uint32_t       bmp_image_size = height * rowSizeNoAlpha;
+    if (bmp_image_size <= data_size)
+    {
+        FILE*   file   = nullptr;
+        int32_t result = util::platform::FileOpen(&file, filename.c_str(), "wb");
+
+        if ((result == 0) && (file != nullptr))
+        {
+            BmpFileHeader file_header;
+            BmpInfoHeader info_header;
+
+            file_header.type      = ('M' << 8) | 'B';
+            file_header.reserved1 = 0;
+            file_header.reserved2 = 0;
+            file_header.off_bits  = sizeof(file_header) + sizeof(info_header);
+            file_header.size      = bmp_image_size + file_header.off_bits;
+
+            info_header.size             = sizeof(info_header);
+            info_header.width            = width;
+            info_header.height           = height;
+            info_header.planes           = 1;
+            info_header.bit_count        = kBmpBitCountNoAlpha;
+            info_header.compression      = 0;
+            info_header.size_image       = 0;
+            info_header.x_pels_per_meter = 0;
+            info_header.y_pels_per_meter = 0;
+            info_header.clr_used         = 0;
+            info_header.clr_important    = 0;
+
+            util::platform::FileWrite(&file_header, sizeof(file_header), 1, file);
+            util::platform::FileWrite(&info_header, sizeof(info_header), 1, file);
+
+            // Y needs to be inverted when writing the bitmap data.
+            auto height_1 = height - 1;
+            auto bytes    = reinterpret_cast<const uint8_t*>(data);
+
+            // Row data without alpha
+            uint8_t* rowBytes = new uint8_t[rowSizeNoAlpha]{ 0 };
```

Comment:
I think if you use an auto `vector` here, you don't have to invoke `new`, and then you don't have to `delete` it - the storage will go away automatically.  So maybe something like `std::vector<uint8_t> rowBytes[rowSizeNoAlpha];` and then use "rowBytes.data()" in `util::platform::FileWrite()`.